### PR TITLE
section-align: AuditLog

### PR DESCRIPTION
**Pull request by peterdrier**
Multi-phase section-align run for the AuditLog section. Phase 0 plan: `docs/plans/2026-05-12-section-align-auditLog.md` (already merged via #498). This PR carries Phase 1–4 execution.

## Phase 1 (this push) — surface alignment

- New `AuditLogController` with `[Route(\"AuditLog\")]` as the front door
- 4 routes moved off `BoardController`/`GoogleController`:
  - `GET /Board/AuditLog` → `GET /AuditLog`
  - `POST /Google/AuditLog/CheckDriveActivity` → `POST /AuditLog/CheckDriveActivity`
  - `GET /Google/Sync/Resource/{id}/Audit` → `GET /AuditLog/Resource/{id}`
  - `GET /Google/Human/{id}/SyncAudit` → `GET /AuditLog/Human/{id}`
- Page views moved: `Views/Shared/AuditLog.cshtml` → `Views/AuditLog/Index.cshtml`; `GoogleSyncAudit.cshtml` → `GoogleSync.cshtml`; section-only partials (`_AuditLogContent`/`_AuditLogScripts`) renamed and moved into the section folder
- ViewModels extracted out of `AdminViewModels.cs` into `AuditLogViewModels.cs`
- Section-specific helper methods moved off `HumansControllerBase` (`GoogleSyncAuditView`, `BuildGoogleSyncAuditViewModel`) into `AuditLogController` as private helpers
- `Extensions/AuditLogUiExtensions.cs` → `Extensions/Sections/AuditLogUiExtensions.cs`
- All inbound links updated (RedirectToAction targets in `GoogleController`, tag-helper links in `Board/Index`, `Google/Index`, `Google/_SyncTabContent`, `Profile/AdminDetail`)

## Phase 2 (next) — arch + boundary protocol

Per plan: lift cross-section reads (`User`/`Team`) out of `AuditLogRepository` into `AuditLogService` calling `IUserService`/`ITeamService`; drop unused `AuditLogEntry.ActorUser` nav; add section-specific + generic arch tests; add interface-method-budget entries; add missing invariant/negative/trigger coverage; split `AuditEventRenderTests`.

## Phase 3 — /simplify

Prune per-section arch test duplicates now covered by generic tests; move display-name helpers private; trim `IAuditLogService` reads.

## Phase 4 — doc polish

Rewrite `docs/sections/AuditLog.md` (Routing, Architecture, Cross-Section Dependencies); update `docs/architecture/dependency-graph.md`.

## Provisional alignment

GoogleIntegration is flagged as the follow-up section-align target — the `AuditLogEntry.Resource` `.Include` and the `DriveActivityMonitorRepository.cs:81` direct DbSet write both depend on GoogleIntegration shipping `IGoogleResourceService.GetByIdsAsync`. After that lands, a follow-up commit returns here to drop the Include and the Resource nav.

## Test plan

- [x] `dotnet build Humans.slnx -v quiet` — 0 errors, 0 warnings
- [x] `dotnet test Humans.slnx -v quiet` (unit tests) — 2,806 pass (Integration suite has pre-existing DB connectivity failures, identical on `main`)
- [ ] Phase 2 adds new arch tests and behavior tests
- [ ] Closing /section-align re-run returns clean (modulo GoogleIntegration follow-up)

### CODE REVIEW

File: src/Humans.Web/Controllers/AuditLogController.cs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+using Humans.Application.Interfaces.Teams;
+using Humans.Application.Services.AuditLog;
+using Humans.Domain.Entities;
+using Humans.Domain.Enums;
+using Humans.Web.Authorization;
+using Humans.Web.Models;
+
+namespace Humans.Web.Controllers;
+
+[Authorize(Policy = PolicyNames.BoardOrAdmin)]
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Web/Controllers/AuditLogController.cs]: Fixed in a1329d637 — dropped the class-level `[Authorize(Policy = BoardOrAdmin)]` and added per-action attributes on `Index`/`CheckDriveActivity`/`Resource`. `Human` keeps its `HumanAdminBoardOrAdmin` and is no longer AND-cumulated with a stricter class-level gate.

File: src/Humans.Application/Services/AuditLog/AuditViewerService.cs [hunk 2]
```diff
@@ -140,6 +141,20 @@ private static AuditEvent Resolve(
             ResourceName: entry.Resource?.Name);
     }
 
+    private async Task<Dictionary<Guid, string>> GetUserDisplayNamesAsync(
+        IReadOnlyList<Guid> userIds, CancellationToken ct)
+    {
+        var users = await _userService.GetByIdsAsync(userIds, ct);
+        return users.ToDictionary(kvp => kvp.Key, kvp => kvp.Value.DisplayName);
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Application/Services/AuditLog/AuditViewerService.cs]: Fixed in a1329d637 — switched display-name resolution from `IUserService.GetByIdsAsync` to `IProfileService.GetByUserIdsAsync`, projecting `Profile.BurnerName`. Entries with null/empty BurnerName are omitted from the dict (existing callers already handle missing names). Doc updated to reflect the dependency swap.

File: src/Humans.Application/Services/AuditLog/AuditLogService.cs [hunk 3]
```diff
@@ -43,17 +44,20 @@ public sealed class AuditLogService : IAuditLogService, IUserDataContributor
 {
     private readonly IAuditLogRepository _repo;
     private readonly IUserService _userService;
+    private readonly ITeamService _teamService;
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Application/Services/AuditLog/AuditLogService.cs]: Fixed in a1329d637 — removed `ITeamService` ctor param, field, assignment, and the `Humans.Application.Interfaces.Teams` using from `AuditLogService`. Test setup updated to match the 4-arg ctor.

File: src/Humans.Infrastructure/Data/Configurations/AuditLog/AuditLogEntryConfiguration.cs [hunk 4]
```diff
@@ -39,8 +39,9 @@ public void Configure(EntityTypeBuilder<AuditLogEntry> builder)
...
             .HasMaxLength(100);
 
-        // FK to User with SetNull on delete (null ActorUserId = system action or deleted user)
-        builder.HasOne(e => e.ActorUser)
+        // FK to User with SetNull on delete (null ActorUserId = system action or deleted user).
+        // No nav property on AuditLogEntry — resolved in-memory via IUserService.GetByIdsAsync.
+        builder.HasOne<User>()
             .WithMany()
             .HasForeignKey(e => e.ActorUserId)
             .OnDelete(DeleteBehavior.SetNull);
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Infrastructure/Data/Configurations/AuditLog/AuditLogEntryConfiguration.cs]: Fixed in a1329d637 — removed the `HasOne<User>()` block from `AuditLogEntryConfiguration` entirely. Added migration `20260512142523_DropAuditLogActorUserIdFk` that drops the DB-level FK constraint + the now-unused index (`DropForeignKey` + `DropIndex` only — snapshot diff verified to touch only the AuditLog HasOne metadata, no collateral). Baseline entry removed; ratchet shrinks as intended. Destructive-ops baseline updated for the two intentional drops.

File: src/Humans.Infrastructure/Migrations/HumansDbContextModelSnapshot.cs [hunk 5]
```diff
@@ -4370,47 +4370,6 @@ protected override void BuildModel(ModelBuilder modelBuilder)
...
-
-                    b.HasIndex("EventSettingsId");
-
-                    b.HasIndex("UserId", "EventSettingsId")
-                        .IsUnique();
-
-                    b.ToTable("volunteer_build_statuses", (string)null);
-                });
-
             modelBuilder.Entity("Humans.Domain.Entities.VolunteerEventProfile", b =>
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Infrastructure/Migrations/HumansDbContextModelSnapshot.cs:4414]: Already fixed in 7530364d (this thread became outdated) — the snapshot was corrupted by a botched `dotnet ef migrations add`/`remove` cycle in Phase 2A and restored from origin/main. Captured the lesson as `memory/process/diff-snapshot-after-ef-tool.md`.

File: src/Humans.Web/Controllers/AuditLogController.cs [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+        var user = await FindUserByIdAsync(id);
+
+        if (user is null)
+        {
+            return NotFound();
+        }
+
+        var events = await _auditViewer.GetGoogleSyncForUserAsync(id);
+        return GoogleSyncAuditView(
+            $"Google Sync Audit: {user.DisplayName}",
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Web/Controllers/AuditLogController.cs]: Fixed in fd815fae3 — injected `IProfileService` and used `GetFullProfileAsync(id)?.DisplayName` (which returns `BurnerName ?? User.DisplayName` per `FullProfile.DisplayName`), with a defensive fallback to `user.DisplayName` for the unexpected null-profile case.

File: src/Humans.Web/Controllers/AuditLogController.cs [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+        if (resource is null)
+        {
+            return NotFound();
+        }
+
+        var events = await _auditViewer.GetForResourceAsync(id);
+        return GoogleSyncAuditView(
+            $"Sync Audit: {resource.Name}",
+            Url.Action("Sync", "Google"),
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Web/Controllers/AuditLogController.cs]: Fixed in b75f268ef — `Url.Action("Sync", "Google")` → `Url.Action(nameof(GoogleController.Sync), "Google")`.

File: src/Humans.Web/Controllers/AuditLogController.cs [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+        {
+            return NotFound();
+        }
+
+        var events = await _auditViewer.GetGoogleSyncForUserAsync(id);
+        var profile = await _profileService.GetFullProfileAsync(id);
+        var displayName = profile?.DisplayName ?? user.DisplayName;
+        return GoogleSyncAuditView(
+            $"Google Sync Audit: {displayName}",
+            Url.Action("AdminDetail", "Profile", new { id }),
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Web/Controllers/AuditLogController.cs]: Fixed in b75f268ef — `Url.Action("AdminDetail", "Profile", ...)` → `Url.Action(nameof(ProfileController.AdminDetail), "Profile", ...)`.

**Comment by peterdrier** [comment]
Fixed in ddb874f77 — dropped `AuditLogEntry.Resource` nav + `HasOne(e => e.Resource)` cross-section mapping; `ResourceId` stays as a bare `Guid?` column. `AuditViewerService` now resolves names via batched `ITeamResourceService.GetResourceNamesByIdsAsync` (the method's producer-side debt remark is also gone). New migration `20260512163538_Drop_AuditLogEntry_ResourceId_FK` is `DropForeignKey`-only; snapshot diff confined to the AuditLog HasOne/Navigation metadata. `NoDestructiveMigrationOps` baseline updated. Scanner-false-negative noted but out of scope for this PR.

File: src/Humans.Infrastructure/Migrations/20260512142523_DropAuditLogActorUserIdFk.cs [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropForeignKey(
+                name: "FK_audit_log_users_ActorUserId",
+                table: "audit_log");
+
+            migrationBuilder.DropIndex(
+                name: "IX_audit_log_ActorUserId",
+                table: "audit_log");
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Infrastructure/Migrations/20260512142523_DropAuditLogActorUserIdFk.cs]: Fixed in c96e41658 (ResourceId FK) + 4d118ed75 (ActorUserId FK + IX). Both migrations and their NoDestructiveMigrationOps baseline entries removed; only code-side drops remain in this PR. Follow-up PR with a hand-written DropForeignKey-only migration will land after #499 soaks in prod.

File: src/Humans.Application/Services/AuditLog/AuditViewerService.cs [hunk 10]
```diff
@@ -137,13 +146,34 @@ private static AuditEvent Resolve(
...
+        IReadOnlyList<Guid> userIds, CancellationToken ct)
+    {
+        // Use IProfileService so BurnerName is returned when the profile has one
+        // (memory/architecture/burnername-is-the-display-name.md). For users
+        // without a profile, or with a null BurnerName (Stub), the entry is
+        // absent from the dict — callers already handle null display names.
+        var profiles = await _profileService.GetByUserIdsAsync(userIds, ct);
+        return profiles
+            .Where(kvp => !string.IsNullOrWhiteSpace(kvp.Value.BurnerName))
+            .ToDictionary(kvp => kvp.Key, kvp => kvp.Value.BurnerName!);
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Application/Services/AuditLog/AuditViewerService.cs:162]: Fixed in 02ec4c73f — dropped the ActorDisplayName/SubjectDisplayName guards in _Content.cshtml; <vc:human> now renders unconditionally when the user-id is present and handles missing-user UX. GetUserDisplayNamesAsync stays as-is (only used by the agent text-render path).

File: src/Humans.Web/Views/Google/_SyncTabContent.cshtml [hunk 11]
```diff
@@ -125,7 +125,7 @@ else
                     @if (Model.CanViewAudit)
                     {
                         <a class="btn btn-sm btn-outline-secondary"
-                           href="@Url.Action("GoogleSyncResourceAudit", "Google", new { id = diff.ResourceId })">Audit</a>
+                           href="@Url.Action("Resource", "AuditLog", new { id = diff.ResourceId })">Audit</a>
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Web/Views/Google/_SyncTabContent.cshtml]: Fixed in 45a5aaf38 — `Url.Action("Resource", "AuditLog", ...)` → `Url.Action(nameof(Humans.Web.Controllers.AuditLogController.Resource), "AuditLog", ...)`. Controllers namespace isn't in _ViewImports so used the FQ path.

File: src/Humans.Infrastructure/Data/Configurations/AuditLog/AuditLogEntryConfiguration.cs [hunk 12]
```diff
@@ -59,12 +53,6 @@ public void Configure(EntityTypeBuilder<AuditLogEntry> builder)
         builder.Property(e => e.UserEmail)
             .HasMaxLength(500);
 
-        // FK to GoogleResource with SetNull on delete
-        builder.HasOne(e => e.Resource)
-            .WithMany()
-            .HasForeignKey(e => e.ResourceId)
-            .OnDelete(DeleteBehavior.SetNull);
-
         // Indexes for querying
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Infrastructure/Data/Configurations/AuditLog/AuditLogEntryConfiguration.cs:68]: Addressed in c919af028 — restored the `HasOne<User>().WithMany().HasForeignKey(e => e.ActorUserId)` and `HasOne<GoogleResource>().WithMany().HasForeignKey(e => e.ResourceId)` blocks plus `HasIndex(e => e.ActorUserId)`. Snapshot now agrees that the FKs/index exist, so the follow-up drop PR can use `dotnet ef migrations add` to auto-generate the DropForeignKey + DropIndex migration after this one soaks in prod — no hand-edited migration required.

File: src/Humans.Web/Controllers/BoardController.cs [hunk 13]
```diff
@@ -21,23 +21,4 @@ public BoardController(
     {
         _auditViewer = auditViewer;
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Web/Controllers/BoardController.cs]: Fixed — `BoardController.cs` deleted entirely (zero remaining actions after `/Board` retired in #507 and AuditLog routes moved into `AuditLogController`). Removed the now-stale `BoardController` row from `EndpointAuthorizationTests.CriticalEndpointPolicies`.

File: src/Humans.Web/Controllers/AuditLogController.cs [hunk 14]
```diff
@@ -0,0 +1,161 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Humans.Application.DTOs;
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Web/Controllers/AuditLogController.cs]: Fixed in 23d0835f1 — dropped the orphan `using Humans.Application.DTOs;` from `AuditLogController.cs`.

File: src/Humans.Application/Interfaces/Tickets/ITicketQueryService.cs [hunk 15]
```diff
@@ -11,7 +10,6 @@ namespace Humans.Application.Interfaces.Tickets;
 /// counts matched tickets, and computes aggregate dashboard statistics.
 /// All matching logic (MatchedUserId, email fallback, case-insensitive) lives here.
 /// </summary>
-[SurfaceBudget(25)]
 public interface ITicketQueryService : IApplicationService
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Application/Interfaces/Tickets/ITicketQueryService.cs:15]: Fixed in 23d0835f1 — restored `ITicketQueryService` and `ITicketTransferService` to their post-#509 state including `[SurfaceBudget(25)]` / `[SurfaceBudget(11)]` and the `using Humans.Application.Architecture;` import. Root cause: the soft-reset-and-recommit step compared against the latest `origin/main` (which had moved forward since the branch's last merge with #509, A2.5a, #510, #511) and silently staged the absent additions as reverts. Restored 12 files via `git checkout origin/main -- …`; amended the commit. Verified `git diff origin/main HEAD --name-only -- src/Humans.Infrastructure/Migrations/ src/Humans.Infrastructure/Data/ src/Humans.Domain/Entities/` is empty, so no DB-schema collateral.

File: tests/Humans.Application.Tests/Authorization/EndpointAuthorizationTests.cs [hunk 16]
```diff
@@ -23,7 +23,6 @@ public class EndpointAuthorizationTests
         { typeof(AdminMergeController), null, "AdminOnly" },
         { typeof(EmailController), null, "AdminOnly" },
         { typeof(AdminController), "Index", "AnyAdminRole" },
-        { typeof(BoardController), null, "BoardOrAdmin" },
         { typeof(GoogleController), "SyncSettings", "AdminOnly" },
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply tests/Humans.Application.Tests/Authorization/EndpointAuthorizationTests.cs:30]: Fixed in 23d0835f1 — added four `AuditLogController` rows to `EndpointAuthorizationTests.CriticalEndpointPolicies`: `Index`, `CheckDriveActivity`, `Resource` (all `BoardOrAdmin`), and `Human` (`HumanAdminBoardOrAdmin`). `Critical_endpoints_require_expected_policy` passes for all four.

**Comment by peterdrier** [comment]
Fixed in 44eafbcbb — `AdminNavTree.cs:48` now points to `("AuditLog", "Index")` instead of the deleted `("Board", "AuditLog")`. Grepped `{*.cs,*.cshtml}` for any other `"Board"` + `"AuditLog"` controller/action pairs — none remaining.
